### ISSUE TITLE
Mapper algorithm using dbscan

### ISSUE THREAD

**Issue by schnirz** [wildart/TDA.jl#6]
The dbscan algorithm is a nice candidate for the clustering algorithm used in Mapper, as it doesn't require an a priori specification of the number of clusters or additional computational steps like elbow detection. An implementation of dbscan in Julia already exists (http://juliastats.github.io/Clustering.jl/stable/dbscan.html), so using dbscan in Mapper should be rather straightforward. 
Thoughts?

**Comment by wildart** [comment]
Good idea. I need to refactor interface to skip the optimal clustering selection step for clustering methods that doesn't rely on it.
